src/js/providers/default.ts

```diff
@@ -18,6 +24,124 @@ export type SeekRange = {
     end: number;
 };
 
+export type ProviderEvents = {
+    [Event.PLAYER_STATE]: {
+        newstate: InternalPlayerState;
+    };
+    [Event.MEDIA_TYPE]: {
+        mediaType: 'audio' | 'video';
+    };
+    [Event.BANDWIDTH_ESTIMATE]: {
+        bandwidthEstimate: number;
+    };
+    [Event.MEDIA_META_CUE_PARSED]: {
+        metadataType: 'media' | 'id3' | 'emsg' | 'date-range' | 'program-date-time' | 'scte-35' | 'discontinuity';
```

Comment:
thoughts on an enum for this type? repeated below

Reply:
Defined a type for it. I don't want to create and use enum const names like MetadataType.MEDIA until we're determined to use them everywhere and that should happen when the commercial providers are types (so that we can enum the events like `Event.MEDIA_META_CUE_PARSED` too).

Reply:
"meta" events can also have a `metadataType` of "unknown". That is assigned to miscellaneous "meta" events from legacy or third party providers like flash that are missing the `metadataType` property.